desktop/libs/aws/src/aws/s3/s3fs.py

```diff
@@ -486,14 +489,24 @@ def _copy(self, src, dst, recursive, use_src_basename):
       if self.isdir(normpath(self.join(S3A_ROOT, key.bucket.name, key.name))):
         dst_name = self._append_separator(dst_name)
 
+      # Set skip_copy to True if the key already exist in destination path and copy is skipped.
+      # This helps later to not the delete the source key then.
+      if dst_bucket.get_key(dst_name, validate=True):
+        skip_copy = True
+
       key.copy(dst_bucket, dst_name)
 
+    return skip_copy
+
   @translate_s3_error
   @auth_error_handler
   def rename(self, old, new):
     new = s3.abspath(old, new)
-    self.copy(old, new, recursive=True)
-    self.rmtree(old, skipTrash=True)
+    skip_copy = self.copy(old, new, recursive=True)
```

Comment:
How about comparing old and new before copying instead? Basically a noop if src and dest is the same..